pkg/chunked/bloom_filter.go

```diff
@@ -0,0 +1,95 @@
+package chunked
+
+import (
+	"encoding/binary"
+	"io"
+	"unsafe"
+)
+
+type bloomFilter struct {
+	bitArray []uint64
+	k        uint32
+}
+
+func newBloomFilter(size uint, k uint32) *bloomFilter {
```

Comment:
(Array indices and sizes in Go are `int`; they can’t go any larger anyway, so using `uint` here only means that the caller has to cast.)